packages/firebase_ui_auth/lib/src/screens/display_name_screen.dart

```diff
@@ -0,0 +1,180 @@
+    this.headerMaxExtent,
+    this.sideBuilder,
+    this.desktopLayoutDirection,
+    this.breakpoint = 500,
+    this.actionCodeSettings,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return FirebaseUIActions(
+      actions: actions,
+      child: UniversalScaffold(
+        body: ResponsivePage(
+          breakpoint: breakpoint,
+          desktopLayoutDirection: desktopLayoutDirection,
+          headerBuilder: headerBuilder,
+          headerMaxExtent: headerMaxExtent,
+          sideBuilder: sideBuilder,
+          maxWidth: 1200,
+          contentFlex: 2,
+          child: Padding(
+            padding: const EdgeInsets.all(32),
+            child: _DisplayNameScreenContent(
+              auth: auth,
+              actionCodeSettings: actionCodeSettings,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+/// This allows a value of type T or T?
+/// to be treated as a value of type T?.
+///
+/// We use this so that APIs that have become
+/// non-nullable can still be used with `!` and `?`
+/// to support older versions of the API as well.
+T? _ambiguate<T>(T? value) => value;
+
+class _DisplayNameScreenContent extends StatefulWidget {
+  /// {@macro ui.auth.auth_controller.auth}
+  final FirebaseAuth? auth;
+  final ActionCodeSettings? actionCodeSettings;
+
+  const _DisplayNameScreenContent({
+    required this.auth,
+    required this.actionCodeSettings,
+  });
+
+  @override
+  State<_DisplayNameScreenContent> createState() =>
+      __DisplayNameScreenContentState();
+}
+
+class __DisplayNameScreenContentState extends State<_DisplayNameScreenContent> {
+  late final controller = EmailVerificationController(auth);
+
+  FirebaseAuth get auth => widget.auth ?? FirebaseAuth.instance;
+
+  User get user => auth.currentUser!;
+  bool isLoading = false;
+
+  @override
+  void initState() {
+    _ambiguate(SchedulerBinding.instance)!
+        .addPostFrameCallback(_sendEmailVerification);
+    super.initState();
+  }
+
+  void _sendEmailVerification(_) {
+    controller
+      ..addListener(() {
+        setState(() {});
+
+        if (state == EmailVerificationState.verified) {
+          final action = FirebaseUIAction.ofType<DisplayNameAction>(context);
+          action?.callback();
+        }
+      })
+      ..sendVerificationEmail(
+        Theme.of(context).platform,
+        widget.actionCodeSettings,
+      );
+  }
+
+  EmailVerificationState get state => controller.state;
+
+  @override
+  Widget build(BuildContext context) {
+    final l = FirebaseUILocalizations.labelsOf(context);
+
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        Align(
+          child: Text(
+            l.name,
+            style: Theme.of(context).textTheme.headline6,
+          ),
+        ),
+        const SizedBox(height: 32),
+        EditableSignUserDisplayName(auth: auth),
+      ],
```

Comment:
This screen needs `continue` and `go back` buttons. With current implementation, if `DisplayNameChanged` action is not provided – the user will be stuck on this screen. I'm doubting that we need this action at all. `continue` and `go back` buttons should be enough.